src/server/tiered_storage.cc

```diff
@@ -359,6 +359,24 @@ void TieredStorage::Free(PrimeIterator it, DbTableStats* stats) {
   stats->tiered_size -= len;
 }
 
+void TieredStorage::Defrag(DbIndex db_index, PrimeIterator it) {
+  PrimeValue& entry = it->second;
+  CHECK(entry.IsExternal());
+  DCHECK_EQ(entry.ObjType(), OBJ_STRING);
+  auto [offset, len] = entry.GetExternalSlice();
+
+  if (offset % kBlockLen > 0) {
+    uint32_t offs_page = offset / kBlockLen;
+    auto refcnt_it = page_refcnt_.find(offs_page);
+    if (refcnt_it->second <= 10) {
```

Comment:
I think that we should check against precentage from NumEntriesInSmallBin.
Because page size is 4096, if the bin size is 2040 than you can add only 2 entries so the count is max 2, but if the bin size is 72 than you can add 51 entries. So I would check if this page is less than 20% utilized than load its entries to memory

Reply:
sounds good